src/core/configuration/configurationService.ts

```diff
@@ -0,0 +1,46 @@
+import { Configuration, WitnessMode } from "./configurationService.types";
+// eslint-disable-next-line no-undef
+const environment = process.env.ENVIRONMENT || "dev-comm";
+
+class ConfigurationService {
+  private static configurationEnv: Configuration;
+  static NOT_FOUND_ENVIRONMENT_FILE =
+    "Can not read/cast environment file, please check your configuration";
+  async start() {
+    await new Promise((rs, rj) => {
+      import(`../../../configs/${environment}.yaml`)
+        .then((module) => {
+          const data = module.default;
+
+          if (this.configurationValid(data)) {
+            ConfigurationService.configurationEnv = data as Configuration;
+          } else {
+            rj(new Error(ConfigurationService.NOT_FOUND_ENVIRONMENT_FILE));
+          }
+
+          rs(true);
+        })
+        .catch(() => {
+          rj(new Error(ConfigurationService.NOT_FOUND_ENVIRONMENT_FILE));
+        });
+    });
+  }
+
+  static get env() {
+    return this.configurationEnv;
+  }
+
+  private configurationValid(data: any): boolean {
+    return (
+      (typeof data.keri === "object" &&
+        typeof data.keri.witness === "string" &&
+        data.keri.witness === WitnessMode.BACKER &&
+        typeof data.keri.backer.address === "string" &&
+        typeof data.keri.backer.aid === "string") ||
+      (data.keri.witness === WitnessMode.POOLS &&
+        Array.isArray(data.keri.pools))
+    );
```

Comment:
This is a bit hard to follow, and the config will only grow but for now lets serialise simply. Also, witness here is a string but actually it's an array of strings.

```
const keri = data.keri;
if (typeof keri !== "object") {
  return false;
}

const backerType = keri.backerType;
if (typeof backerType !== "string") {
  return false;
}

if (backerType === WitnessMode.POOLS) {
  const witnessConfig = keri.pools;
  return Array.isArray(witnessConfig) && witnessConfig.length > 0 && witnessConfig.every(witness => typeof witness === "string");
} else if (backerType === WitnessMode.LEDGER) {
  const ledgerConfig = keri.ledger;
  if (typeof ledgerConfig !== "object") {
    return false;
  }

  return typeof ledgerConfig.aid === "string" && typeof ledgerConfig.address === "string";
} else {
  return false;
}
```

Ideally we should have some serialise/deserialise code so that we can better handle this with annotations or something but lets not introduce another dependency just for this. We can re-visit to make this more readable.

NOTE: I just typed this here, it mightn't be 100% correct.